src/sentry/api/serializers/models/group.py

```diff
@@ -405,12 +405,24 @@ def _get_permalink(self, obj, user):
                 request.auth, obj.organization
             )
 
-        if (
-            is_superuser
-            or (user.is_authenticated() and user.get_orgs().filter(id=obj.organization.id).exists())
-            or is_valid_sentryapp
-        ):
-            return obj.get_absolute_url()
+        with sentry_sdk.start_span(op="GroupSerializerBase.serialize.permalink.check") as span:
+
+            def user_is_in_org():
+                exists = user.get_orgs().filter(id=obj.organization.id).exists()
+                span.set_data("user_is_in_org", exists)
+                return exists
+
+            permalink_is_visible = (
+                is_superuser or (user.is_authenticated() and user_is_in_org()) or is_valid_sentryapp
```

Comment:
At a glance, this might be optimized by rearranging to
`is_superuser or is_valid_sentryapp or (user.is_authenticated() and user_is_in_org())`.
But I'd first like to capture some data on whether we'd expect it to make a difference.